src/ui/public/timefilter/timefilter.js

```diff
@@ -23,6 +23,12 @@ uiModules
   var Events = Private(EventsProvider);
   var diff = Private(UtilsDiffTimePickerValsProvider);
 
+  config.$bind($rootScope, 'dateFormat:dow', 'dateFormat_dow');
+
+  $rootScope.$watch('dateFormat_dow', function (day) {
```

Comment:
The global one is a wrapper for the regular one. kbnGlobalTimepicker is what you put to make the time showup in the corner.